design/key_derivation.md

```diff
@@ -0,0 +1,42 @@
+# Key Derivation design
+
+## Scope
+
+Obscuro blockchain is seeded with a network Key.
+This network key is shared between enclaves upon successful attestation.
+
+Transactions are encrypted in the rollups that make the Obscuro blockchain.
+Each rollup uses a key to encrypt the Transactions.
+This key is derived from the network Key.
+
+## Requirements
+
+* Each rollup is encrypted with a unique key
+  * Each unique key is derived from the network key
+  * Each unique key is not able to disclose other unique keys
+  * Each unique key is not able to disclose the network key
+  * Each unique key is deterministic give the rollup height
+* Each enclave is able to determine the unique key given the enclave secret and rollup height
```

Comment:
the requirement is to encrypt with symmetric keys.
Also need a bit of analysis of the existing symmetric algorithms.

Reply:
Added more info 👍 

I think for the first iteration AES is the right tool. It's the industry standard.
I did do a more in-depth on what type of cipher modes we should use. I think that's pretty interesting.